web/src/message_list.js

```diff
@@ -344,7 +344,7 @@ export class MessageList {
         const is_web_public = sub.is_web_public;
         const can_toggle_subscription =
             sub !== undefined && stream_data.can_toggle_subscription(sub);
-        if (sub === undefined) {
+        if (sub.archived || sub === undefined) {
```

Comment:
With deactivated streams now being returned to the client, could sub be `undefined`? If so, should we assume it's deactivated?

Also we should rename `deactivated` to `archived` maybe but that should be done in a follow up PR ig.

Reply:
I think it is still possible for `sub` to be undefined, but only in the preserve of bugs. Probably a follow-up project for sure.

And yes, we should plan to do follow-up passes renaming `deactivated to `archived`, both here and in the server implementation -- but each should be its own project, since the server one will require a database migration (and thus a lot of work), while the web app rename should be an easy change.